File: tactode-nodejs/params.js
const fs = require('fs')

// GLOBAL VARIABLES
const TESTS = {
  config: 'testing/config_tests_js.txt',
  tiles: 'testing/pieces.txt'
}

const RESULTS = {
  folder: '../../results/',
  file: 'results.txt'
}

const DATASET = {
  templates: '../dataset/templates',
  test_small: '../dataset/test/tactode_small',
  test_big: '../dataset/test/tactode_big',
  train_cla: '../dataset/train/tile_classification',
  train_det: '../dataset/train/teeth_detection'
}

// CLASSES
class PipelineResult {
  constructor (rects, src, isRotated) {
    this.rectsArr = rects
    this.srcImg = src
    this.isRotated = isRotated
  }
}

class ClassificationResult {
  constructor (tiles = [], numTiles = -1, isRotated = false) {
    this.tilesArr = tiles
    this.numberOfTiles = numTiles
    this.isRotated = isRotated
  }
}

class Size {
  constructor (w = -1, h = -1) {
    this.width = w
    this.height = h
  }
}

class HogSvmDetector {
  constructor () {
    this.svmFile = '../trained_models/hogsvm_JS/svm_detector_JS.yaml'
    this.hogFile = '../trained_models/hogsvm_JS/hog_detector.yaml'
    this.size = new Size(64, 32)
  }
}

class HogSvmClassifier {
  constructor () {
    this.classesFile = '../trained_models/hogsvm_JS/pieces.txt'
    this.svmFile = '../trained_models/hogsvm_JS/svm_classifier_JS.yaml'
    this.hogFile = '../trained_models/hogsvm_JS/hog_classifier.yaml'
    this.size = new Size(32, 32)
    this.classes = []
  }

  loadClasses () {
    this.classes = fs.readFileSync(this.classesFile).toString().split('\n')
  }
}

class TemplateMatchFeatDetDesc {
  constructor () {
    this.folder = DATASET.templates
    this.size = new Size(350, 350)
  }
}

// FUNCTIONS
function getCurrentTime () {
  const date = new Date()
  let dayOfMonth = date.getDate(); dayOfMonth = (dayOfMonth < 10) ? `0${dayOfMonth}` : dayOfMonth
  let month = date.getMonth(); month = (month < 10) ? `0${month + 1}` : month + 1
  let year = date.getFullYear(); year = (year < 10) ? `0${year}` : year
  let hours = date.getHours(); hours = (hours < 10) ? `0${hours}` : hours
  let minutes = date.getMinutes(); minutes = (minutes < 10) ? `0${minutes}` : minutes
  let seconds = date.getSeconds(); seconds = (seconds < 10) ? `0${seconds}` : seconds
  return `${dayOfMonth}/${month}/${year} ${hours}:${minutes}:${seconds}`
}

module.exports = {
  TESTS,
  RESULTS,
  DATASET,
  PipelineResult,
  ClassificationResult,
  Size,
  HogSvmDetector,
  HogSvmClassifier,
  TemplateMatchFeatDetDesc,
  getCurrentTime
}

File: tactode-python/params.py
import time

# GLOBAL VARIABLES
TESTS = {
    'config': 'testing/config_tests_py.txt',
    'tiles': 'testing/tiles.txt'
}

RESULTS = {
    'folder': '../../results/',
    'file': 'results.txt'
}

DATASET = {
    'templates':  '../dataset/templates',
    'test_small': '../dataset/test/tactode_small',
    'test_big':   '../dataset/test/tactode_big',
    'train_cla':  '../dataset/train/tile_classification',
    'train_det':  '../dataset/train/teeth_detection'
}


# CLASSES
class ConfMatrix:
    def __init__(self, name=""):
        self.id = name
        self.count = 0


class Confusions:
    def __init__(self, name=""):
        self.id = name
        self.tp = 0
        self.fp = 0
        self.fn = 0
        self.conf_matrix = []


class PieceError:
    def __init__(self, row=-1, col=-1, expected_p="", obtained_p=""):
        self.i = row
        self.j = col
        self.expected_piece = expected_p
        self.obtained_piece = obtained_p


class ErrorsList:
    def __init__(self, lst=[], n_err=-1):
        self.list = lst
        self.num_errors = n_err


class PiecesList:
    def __init__(self, lst=[], n_pieces=-1, rot=False):
        self.list = lst
        self.num_pieces = n_pieces
        self.is_rotated = rot


class Size:
    def __init__(self, w=-1, h=-1):
        self.width = w
        self.height = h


class HogSvmDetector:
    def __init__(self):
        self.svm_file = "../trained_models/hogsvm_PY/svm_detector_PY.yaml"
        self.hog_file = "../trained_models/hogsvm_PY/hog_detector.yaml"
        self.size = Size(w=64, h=32)


class HogSvmClassifier:
    def __init__(self):
        self.classes_file = "../trained_models/hogsvm_PY/pieces.txt"
        self.svm_file = "../trained_models/hogsvm_PY/svm_classifier_PY.yaml"
        self.hog_file = "../trained_models/hogsvm_PY/hog_classifier.yaml"
        self.size = Size(w=32, h=32)
        self.classes = ""

    def load_classes(self):
        with open(self.classes_file, "r") as f:
            self.classes = f.readlines()


class YOLO:
    def __init__(self, obj_thresh=.1, conf_thresh=.1, nms_thresh=.1):
        self.classes_file = "../trained_models/yolo/obj.names"
        self.cfg_file = "../trained_models/yolo/yolov4-obj.cfg"
        self.weights_file = "../trained_models/yolo/yolov4-obj_last.weights"
        self.size = Size(w=416, h=416)
        self.object_threshold = obj_thresh  # Objects threshold
        self.conf_threshold = conf_thresh   # Confidence threshold
        self.nms_threshold = nms_thresh     # Non-maximum suppression threshold
        self.classes = ""

    def load_classes(self):
        with open(self.classes_file, "rt") as f:
            self.classes = f.read().rstrip("\n").split("\n")


class SSD:
    def __init__(self, obj_thresh=.1, conf_thresh=.1, nms_thresh=.1):
        self.classes_file = "../trained_models/ssd/classes.txt"
        self.model_file = "../trained_models/ssd/frozen_inference_graph.pb"
        self.size = Size(w=300, h=300)
        self.object_threshold = obj_thresh
        self.conf_threshold = conf_thresh
        self.nms_threshold = nms_thresh
        self.classes = ""

    def load_classes(self):
        with open(self.classes_file, "rt") as f:
            self.classes = f.read().rstrip("\n").split("\n")


class VGG16:
    def __init__(self):
        self.classes_file = "../trained_models/vgg16/classes.txt"
        self.model_folder = "../trained_models/vgg16/finetuned"
        self.size = Size(w=224, h=224)
        self.classes = ""

    def load_classes(self):
        with open(self.classes_file, "rt") as f:
            self.classes = f.read().rstrip("\n").split("\n")


class VGG19:
    def __init__(self):
        self.classes_file = "../trained_models/vgg19/classes.txt"
        self.model_folder = "../trained_models/vgg19/finetuned"
        self.size = Size(w=224, h=224)
        self.classes = ""

    def load_classes(self):
        with open(self.classes_file, "rt") as f:
            self.classes = f.read().rstrip("\n").split("\n")


class MobileNetV2:
    def __init__(self):
        self.classes_file = "../trained_models/mobilenetV2/classes.txt"
        self.model_folder = "../trained_models/mobilenetV2/finetuned"
        self.size = Size(w=224, h=224)
        self.classes = ""

    def load_classes(self):
        with open(self.classes_file, "rt") as f:
            self.classes = f.read().rstrip("\n").split("\n")


class TemplateMatchFeatDetDesc:
    def __init__(self):
        self.folder = DATASET['templates']
        self.size = Size(w=350, h=350)


# FUNCTIONS
def get_current_time():
    return time.strftime("%d/%m/%Y %H:%M:%S", time.localtime())
